libsolidity/formal/CHC.cpp

```diff
@@ -75,16 +77,42 @@ void CHC::analyze(SourceUnit const& _source)
 	m_context.setAssertionAccumulation(false);
 	m_variableUsage.setFunctionInlining(false);
 
+	resetSourceAnalysis();
+
 	auto boolSort = make_shared<smt::Sort>(smt::Kind::Bool);
 	auto genesisSort = make_shared<smt::FunctionSort>(
 		vector<smt::SortPointer>(),
 		boolSort
 	);
 	m_genesisPredicate = createSymbolicBlock(genesisSort, "genesis");
-	auto genesis = (*m_genesisPredicate)({});
-	addRule(genesis, genesis.name);
+	addRule(genesis(), "genesis");
+
+	defineInterfacesAndSummaries(_source);
+	for (auto const& source: _source.referencedSourceUnits())
+		defineInterfacesAndSummaries(*source);
 
 	_source.accept(*this);
+	while (!m_toAnalyzeContracts.empty())
```

Comment:
Currently it's empty. It's filled when there's a function call to a contract that is not `m_currentContract` or any of its bases.

Reply:
Ok - so it might make sense to remove it here and move it to the PR that introduces that... but I'd also be fine with keeping it here, if it'd be added by the next PR anyways.